src/main/java/by/itech/lab/supplier/service/impl/CustomerServiceImpl.java

```diff
@@ -29,4 +37,42 @@ public CustomerDto getCustomer(final Long customerId) {
                 .orElseThrow(() -> new ResourceNotFoundException("Customer with id=" + customerId + " doesn't exist"));
     }
 
+    @Override
+    public CustomerDto save(CustomerDto customerDto) {
+        Customer customer = Optional.ofNullable(customerDto.getId())
+                .map(item -> {
+                    final Customer existing = customerRepository
+                            .findById(customerDto.getId())
+                            .orElseThrow();
+                    customerMapper.update(customerDto, existing);
+                    return existing;
+                })
+                .orElseGet(() -> customerMapper.map(customerDto));
+
+        final Customer saved = customerRepository.save(customer);
+        userService.save(createAdmin());
+        return customerMapper.map(saved);
+    }
+
+    @Override
+    public List<Customer> saveNewStatus(List<CustomerDto> customerDtoList) {
```

Comment:
maybe change only status, not all entity

Reply:
good